angel_system/data/recipe_task_graph.py

```diff
@@ -30,59 +47,84 @@ def load_recipe(recipe):
 
 
 def get_all_recipe_orders(labels):
+    """
+    Creates a recipe graph and determines all
+    possible topological sortings of the graph
+
+    :param labels: List of activity labels
+
+    :return: A tuple consisting of two elements:
+        0: List of all possible topological sortings of the recipe graph
+        1: List of ids that represent the root node of a set of repeated actions
+    """
     G = nx.DiGraph()
+    repeated_ids = []
 
     for action in labels:
-        act_label = action["label"]
-        depends = action["depends"]
+        act_id = action["id"]
+        G.add_node(act_id)
+
+        if act_id == 0:
+            continue
+
+        depends = action.get("depends", None)
 
         if "repeat" in action.keys():
             repeat = action["repeat"]
+            repeated_ids.append(act_id)
 
-            # Node that indicated when the full repetition is finished
-            all_act_label = f"{act_label}-all"
-            G.add_node(all_act_label)
             for r in range(repeat):
-                new_act_label = f"{act_label}-{r}"
-                G.add_edge(new_act_label, all_act_label)
+                new_act_id = f"{act_id}-{r}"
+                G.add_edge(new_act_id, act_id)
 
-                G.add_node(new_act_label, root_label=act_label)
+                G.add_node(new_act_id, root_label=act_id)
                 if depends is not None:
                     for d in depends:
-                        G.add_edge(d, new_act_label)
+                        G.add_edge(d, new_act_id)
+                else:
+                    G.add_edge(0, new_act_id)
         else:
-            G.add_node(act_label)
-
             if depends is not None:
                 for d in depends:
-                    G.add_edge(d, act_label)
+                    G.add_edge(d, act_id)
+            else:
+                G.add_edge(0, act_id)
+
+    nx.draw(G, with_labels=True)
+    plt.savefig("coffee.png")
```

Comment:
Is this debugging code that should be removed? Either way, the "coffee" labeling here is inappropriate hard coding. Maybe this is for another branch at this stage.